### PR TITLE
More codegen cleanups

### PR DESCRIPTION
**Pull request by nnethercote**
Some additional cleanups I found while looking closely at this code, following up from #112827.

r= @oli-obk

### CODE REVIEW

File: compiler/rustc_interface/src/queries.rs [hunk 1]
```diff
@@ -323,7 +317,7 @@ impl<'tcx> Queries<'tcx> {
             dep_graph,
             prepare_outputs,
             crate_hash,
-            ongoing_codegen,
+            ongoing_codegen: ongoing_codegen,
```
Comment (oli-obk) [inline compiler/rustc_interface/src/queries.rs:320]: This diff is not necessary

**Review by oli-obk — CHANGES_REQUESTED** [review]
r=me with nit

File: compiler/rustc_interface/src/queries.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_interface/src/queries.rs]: :+1:

**Comment by nnethercote** [comment]
The first commit caused the crash, and I don't understand why. I have removed it, I'll try doing it in a separate PR. The other two commits can be merged here.

@bors r=oli-obk rollup

**Comment by nnethercote** [comment]
> The first commit caused the crash, and I don't understand why. I have removed it, I'll try doing it in a separate PR.

It was because I changed the type of `__rustc_codegen_backend` but didn't also change its callers. (This isn't caught by the type checker because of the code structure.) I found a simpler way to do it that doesn't change the type in #112913.

**Pull request merged**